testcases/kernel/syscalls/write/write05.c

```diff
@@ -91,6 +91,15 @@ static void verify_write(void)
 	if (fd < 0)
 		tst_res(TFAIL, "creating a new file failed");
 
+//block2.1:
+	if (write(fd, NULL, 0) != 0) {
+		tst_res(TFAIL, "write() with nsize 0 in a "
+				"valid fd failed with errno: %d, "
+				"but should have succeeded", errno);
```

Comment:
Just do tst_res(TFAIL | TERRNO, ...) instead of printing the value by hand, that way the user will see also the actual errno name not just the number.